CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h

```diff
@@ -23,8 +24,9 @@ namespace reco {
   template<typename Object>
   struct ValueOnObject {
     virtual double eval(Object const&) const =0;
+    virtual ~ValueOnObject(){};
```

Comment:
glimps shows that these classes are not use (at least in cmssw).
```
>cmsglimpse CutOnObject
src/CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h:   struct CutOnObject {
>cmsglimpse SelectIndecesInCollection
src/CommonTools/Utils/interface/ExpressionEvaluatorTemplates.h:   struct SelectIndecesInCollection {
```
@mrodozov , I would suggest to just add the missing destructor.